### PR TITLE
Get a WebXR device registry to each script thread

### PR DESCRIPTION
**Pull request by asajeffrey**
This PR gets an XR device registry to each script thread.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because it's all plumbing

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23690)

### CODE REVIEW

**Comment by asajeffrey** [comment]
Depends on #23675 and https://github.com/servo/webxr/pull/3

File: Cargo.toml [hunk 1]
```diff
@@ -29,3 +29,7 @@ opt-level = 3
 mio = { git = "https://github.com/servo/mio.git", branch = "servo" }
 iovec = { git = "https://github.com/servo/iovec.git", branch = "servo" }
 cmake = { git = "https://github.com/alexcrichton/cmake-rs" }
+
+[patch."https://github.com/servo/webxr"]
+webxr = { path = "../webxr/webxr" }
```
Comment (Manishearth) [inline Cargo.toml]: nope

**Review by Manishearth — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (asajeffrey) [reply Cargo.toml]: Fixed.

**Pull request merged**